pkg/errors/errors.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+// The idea of writing errors package in thanos is highly motivated from the Tast project of Chromium OS Authors. However, instead of
+// copying the package, we end up writing our own simplified logic borrowing some ideas from the errors and github.com/pkg/errors.
+// A big thanks to all of them.
+
+// Package errors provides basic utilities to manipulate errors with a useful stacktrace. It combines the
+// benefits of errors.New and fmt.Errorf world into a single package.
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// base is the fundamental struct that implements the error interface and the acts as the backbone of this errors package.
+type base struct {
+	info  string     // error message passed through calls like errors.Wrap, errors.Errorf
+	stack stacktrace // stacktrace where this error was generated
+	err   error      // actual error which
+}
+
+// Error implements the error interface.
+func (b *base) Error() string {
+	if b.err != nil {
+		return fmt.Sprintf("%s: %s", b.info, b.err.Error())
+	}
+	return b.info
+}
+
+// Unwrap implements the error Unwrap interface.
+func (b *base) Unwrap() error {
+	return b.err
+}
+
+// Format implements the fmt.Formatter interface to support the formatting of an error chain with "%+v" verb.
+func (b *base) Format(s fmt.State, verb rune) {
+	if verb == 'v' && s.Flag('+') {
+		s.Write([]byte(formatErrorChain(b)))
+		return
+	}
+
+	s.Write([]byte(b.Error()))
+}
+
+// New creates a new error with the given message and a stacktrace in details.
+// An alternative to errors.New function.
+func New(msg string) error {
+	return &base{
+		info:  msg,
+		stack: newStackTrace(),
+		err:   nil,
+	}
+}
+
+// Errorf creates a new error with the given message and a stacktrace in details.
+// An alternative to fmt.Errorf function.
+func Errorf(format string, args ...interface{}) error {
```

Comment:
Same here - again - it's our package, we can innovate. 

What about only 

`errors.Newf` and `errors.Wrapf`?

Or even naming it without `f` but supporting fmt? No harm here, but simper API for everyone. So:

`errors.New(format string, args ...interface{})` and `errors.Wrap(err, format string, args ...interface{})` would work too

Reply:
Nice.

After some thinking, let's keep `f` word suffix so we are consistent with some lints/tooling that tries to verify formatters. Just removing non f and being able to produce `format` string without variables is fine. WDYT?

Reply:
Also next time, just in case use 

```
func benchWrapf(errorMsg *string, b *testing.B) {
       b.ResetAllocs()
	for i := 0; i < b.N; i++ {
		_ = errors.Wrap(errors.New("random error"), *errorMsg)
	}
}
```

Reply:
Great! Thanks